internal/storage/pg/pg.go

```diff
@@ -68,27 +74,108 @@ func (s *SQLStorage) PutBatchURLs(ctx context.Context, urls []storage.URLRecord)
 	return tx.Commit()
 }
 
-func (s *SQLStorage) GetURL(ctx context.Context, shortURL string) (*storage.URLRecord, error) {
+func (s *SQLStorage) GetURL(ctx context.Context, shortURL string) (string, error) {
 	timeoutCtx, cancel := context.WithTimeout(ctx, time.Second*3)
 	defer cancel()
 
 	query := `
-		SELECT id, short_url, original_url 
+		SELECT original_url, is_deleted
 		FROM short_urls
 		WHERE short_url = $1`
 
 	row := s.db.QueryRowContext(timeoutCtx, query, shortURL)
 
-	var url storage.URLRecord
-	if err := row.Scan(&url.UUID, &url.ShortURL, &url.OriginalURL); err != nil {
-		return nil, err
+	var originalURL string
+	var isDeleted bool
+	if err := row.Scan(&originalURL, &isDeleted); err != nil {
+		return "", err
 	}
 
 	if err := row.Err(); err != nil {
+		return "", err
+	}
+
+	if isDeleted {
+		return "", ErrURLDeleted
+	}
+
+	return originalURL, nil
+}
+
+func (s *SQLStorage) GetURLsByUserID(ctx context.Context, userID string) ([]*storage.URLRecord, error) {
+	timeoutCtx, cancel := context.WithTimeout(ctx, time.Second*3)
+	defer cancel()
+
+	query := `
+		SELECT id, user_id, short_url, original_url 
+		FROM short_urls
+		WHERE user_id = $1`
+
+	rows, err := s.db.QueryContext(timeoutCtx, query, userID)
+	if err != nil {
 		return nil, err
 	}
+	defer rows.Close()
 
-	return &url, nil
+	var records []*storage.URLRecord
+	for rows.Next() {
+		var r storage.URLRecord
+		if err := rows.Scan(&r.UUID, &r.UserID, &r.ShortURL, &r.OriginalURL); err != nil {
+			return nil, err
+		}
+
+		records = append(records, &r)
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, err
+	}
+
+	if len(records) == 0 {
+		return nil, fmt.Errorf("no short urls for id %s", userID)
+	}
+
+	return records, nil
+}
+
+func (s *SQLStorage) CheckExistence(ctx context.Context, shortURL, userID string) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, time.Second*3)
+	defer cancel()
+
+	query := `
+		SELECT id 
+		FROM short_urls
+		WHERE user_id = $1 AND short_url = $2`
+
+	row := s.db.QueryRowContext(timeoutCtx, query, userID, shortURL)
+
+	var id string
+	if err := row.Scan(&id); err != nil {
+		return err
+	}
+
+	if err := row.Err(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SQLStorage) DeleteURLBatch(urls []string, user string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+	defer cancel()
+
+	urlsString := "{" + strings.Join(urls, ",") + "}"
+
+	query := `
+		UPDATE short_urls 
+		SET is_deleted = true 
+		WHERE short_url = ANY($1::text[]) AND user_id = $2::uuid`
```

Comment:
можно использовать конструкцию 
```suggestion
		WHERE short_url IN(1,2,3,4) AND user_id = $2::uuid`
```